packages/firebase_auth/firebase_auth/ios/Classes/PigeonParser.m

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+#import "PigeonParser.h"
+#import <Foundation/Foundation.h>
+
+@implementation PigeonParser
+
++ (PigeonUserCredential *)getPigeonUserCredentialFromAuthResult:
+    (nonnull FIRAuthDataResult *)authResult {
+  return [PigeonUserCredential
+            makeWithUser:[self getPigeonDetails:authResult.user]
+      additionalUserInfo:[self getPigeonAdditionalUserInfo:authResult.additionalUserInfo]
+              credential:[self getPigeonAuthCredential:authResult.credential]];
+}
+
++ (PigeonUserCredential *)getPigeonUserCredentialFromFIRUser:(nonnull FIRUser *)user {
+  return [PigeonUserCredential makeWithUser:[self getPigeonDetails:user]
+                         additionalUserInfo:nil
+                                 credential:nil];
+}
+
++ (PigeonUserDetails *)getPigeonDetails:(nonnull FIRUser *)user {
+  return [PigeonUserDetails makeWithUserInfo:[self getPigeonUserInfo:user]
+                                providerData:[self getProviderData:user.providerData]];
+}
+
++ (PigeonUserInfo *)getPigeonUserInfo:(nonnull FIRUser *)user {
+  return [PigeonUserInfo
+              makeWithUid:user.uid
+                    email:user.email
+              displayName:user.displayName
+                 photoUrl:(user.photoURL.absoluteString.length > 0) ? user.photoURL.absoluteString
+                                                                    : nil
+              phoneNumber:user.phoneNumber
+              isAnonymous:[NSNumber numberWithBool:user.isAnonymous]
+          isEmailVerified:[NSNumber numberWithBool:user.emailVerified]
+               providerId:user.providerID
+                 tenantId:user.tenantID
+             refreshToken:user.refreshToken
+        creationTimestamp:@((long)([user.metadata.creationDate timeIntervalSince1970] * 1000))
+      lastSignInTimestamp:@((long)([user.metadata.lastSignInDate timeIntervalSince1970] * 1000))];
+}
+
++ (NSArray<NSDictionary<id, id> *> *)getProviderData:
+    (nonnull NSArray<id<FIRUserInfo>> *)providerData {
+  NSMutableArray<NSDictionary<id, id> *> *dataArray =
+      [NSMutableArray arrayWithCapacity:providerData.count];
+
+  for (id<FIRUserInfo> userInfo in providerData) {
+    NSDictionary *dataDict = @{
+      @"providerId" : userInfo.providerID,
+      // Can be null on emulator
+      @"uid" : userInfo.uid ?: @"",
+      @"displayName" : userInfo.displayName ?: [NSNull null],
+      @"email" : userInfo.email ?: [NSNull null],
+      @"phoneNumber" : userInfo.phoneNumber ?: [NSNull null],
+      @"photoURL" : userInfo.photoURL.absoluteString ?: [NSNull null],
+      @"isAnonymous" : @NO,
+      @"isEmailVerified" : @YES,
```

Comment:
Should these 2 BOOLs be hard-coded?

Reply:
In Provider Data, the email is always verified and the user is never anonymous. I'm adding a comment